clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -637,6 +638,9 @@ private void handleResponse(RequestHeader header, Struct body, long now) {
                 log.trace("Ignoring empty metadata response with correlation id {}.", header.correlationId());
                 this.metadata.failedUpdate(now);
             }
+            for (String topic : response.errorTopics(Errors.UNKNOWN_TOPIC_OR_PARTITION))
+                this.metadata.maybeExpire(topic, now);
```

Comment:
@MayureshGharat Thank you for the review. I have renamed the method. Yes, I will add some unit tests later today (wanted some feedback on the code first to make sure it made sense).